CMakeLists.txt

```diff
@@ -102,6 +102,14 @@ Cabana_add_dependency( PACKAGE HYPRE VERSION 2.22.1 )
 # find heffte
 Cabana_add_dependency( PACKAGE Heffte VERSION 2.1.0 )
 
+# find Silo
+Cabana_add_dependency( PACKAGE SILO )
+if(SILO_FOUND)
+  install(FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindSILO.cmake
```

Comment:
The development version of SILO seems to be exporting the package configuration with CMake.  https://github.com/LLNL/Silo/blob/a1e27d34e8b52196be2de3d4aae2d0cae21b3f36/CMakeLists.txt#L675-L706

Do we actually need to write our own find module?

Reply:
@markcmiller86: Does silo have a CMake find module?

Reply:
@junghans I don't know but @biagas would. I know @biagas is just adding CMake build functionality to Silo but I honestly don't know if that included its own FindSilo functionality to be exported for other projects to use. @biagas, do you have any comments/thoughts here?

Reply:
The cmake build and package config exports is new and not yet well tested on all platforms, so it is still considered experimental at this stage, especially the exported cmake configs.

That said, the current state of things regarding CMake and Silo:

The CMake build should work on linux and Windows.

There isn't a find module for Silo. CMake should be able to find it with "find_package"' if Silo_DIR is set to location of Silo's install cmake subdir (eg location of SiloConfig.cmake) before the call to find_package.  Alternatively, including the SiloConfig.cmake file directly should work.

If Silo was built with HDF5 support, path to HDF5 should also be specified before the call to find_package for Silo, or perform a Find for HDF5 before Silo.

Here's a sample CMakeLists.txt I used for testing find_package with Silo (on Windows)

```
cmake_minimum_required(VERSION 3.14)

project(TestImportingSilo)

# path to Silo
set(Silo_DIR "/path/to/installed_silo/cmake")

# for HDF5 built with CMake, SiloConfig.cmake will use this to find hdf5
set(HDF5_DIR "/path/to/installed_hdf5/cmake")

find_package(Silo REQUIRED CONFIG)

if(Silo_FOUND)
    message("Silo_FOUND")
else()
    message("NOT Silo_FOUND")
endif()

```

Reply:
@biagas does it matter how we build silo or do we need to build silo with CMake to get `SiloConfig.cmake`.

Reply:
@junghans I just wanna make sure you are aware that the CMake work @biagas has completed so far is not in any released version of Silo yet. Its on our [`main` branch](https://github.com/LLNL/Silo/commits/main)

Reply:
For now we plan to support non-CMake builds and switch to CMake-only when released - thanks for all the details

Reply:
We currently add and link `SILO::silo` - @biagas will that break compatibility with the current/planned CMake support?

Reply:
We haven't namespaced silo in the CMake support yet, but it makes a lot of sense to do so. 

Shouldn't be a problem. I've created a [ticket](https://github.com/LLNL/Silo/issues/242) to ensure the exported targets are namespaced with 'SILO'.

Reply:
Thanks!